### PR TITLE
528 reuseable generate test cases

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #528

- [x] Tests
- ~~Layman Test Client (including docker image at docker hub)~~
- ~~Changelog~~
- ~~Documentation~~

### CODE REVIEW

File: tests/dynamic_data/base_test_util.py [hunk 1]
```diff
@@ -106,7 +107,14 @@ def check_specific_parametrizations(rest_parametrization, specific_parametrizati
...
             dimension_enum = get_dimension_enum(dimension)
-            param_values = [v for v in sp_parametrization if v in dimension_enum]
+            param_values = []
+            for parametrization_key in sp_parametrization:
+                if inspect.isclass(parametrization_key) and issubclass(parametrization_key, Enum):
+                    if parametrization_key == dimension_enum:
+                        param_values.append(parametrization_key)
+                else:
+                    if parametrization_key in dimension_enum:
+                        param_values.append(parametrization_key)
```
Comment (jirik) [inline tests/dynamic_data/base_test_util.py:117]: if parametrization_key == dimension_enum:
                    param_values.append(parametrization_key)
                elif parametrization_key in dimension_enum:
                    param_values.append(parametrization_key)

param_values = [v for v in sp_parametrization if v == dimension_enum or v in dimension_enum]

File: tests/dynamic_data/base_test.py [hunk 2]
```diff
@@ -91,6 +91,12 @@ def parametrize_test_cases(cls) -> [TestCaseType]:
 
         test_cases = []
         for input_test_case in cls.test_cases:
+            specific_types = {}
+            for source_filter, test_type in input_test_case.specific_types.items():
+                filter_set = util.case_to_simple_parametrizations(source_filter)
+                for filter in filter_set:
+                    assert filter not in specific_types
+                    specific_types[filter] = test_type
```
Comment (jirik) [inline tests/dynamic_data/base_test.py:99]: Encapsulate it to method and use the method for specific_types and specific_params.

**Review by jirik — CHANGES_REQUESTED** [review]

File: tests/dynamic_data/base_test.py (hunk 2 above)
Comment (jirik) [reply tests/dynamic_data/base_test.py:99]: Also add unit tests (positive and negative)

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — APPROVED** [review]
